include/aws/io/io.h

```diff
@@ -14,12 +14,36 @@ AWS_PUSH_SANE_WARNING_LEVEL
 
 #define AWS_C_IO_PACKAGE_ID 1
 
+struct aws_io_handle;
+
+#if AWS_USE_ON_EVENT_WITH_RESULT
+struct aws_event_loop;
+
+/**
+ * Results of the I/O operation(s) performed on the aws_io_handle.
+ */
+struct aws_io_handle_io_op_result {
+    size_t read_bytes;
+    size_t written_bytes;
+    int read_error_code;
```

Comment:
What is the domain here?  crt error codes or os error codes?

Reply:
Yes, this field is a CRT error code (e.g. `AWS_IO_READ_WOULD_BLOCK`). I can rename it to `read_crt_error_code`, wdyt?

Reply:
Probably fine as is.